specs/rigil/precompiles.md

```diff
@@ -97,82 +131,124 @@ Retrieves data from the confidential store. Also mandates the caller's presence
 ```
 
-### `NewDataRecord`
+Inputs:
 
-[🔗 Implementation](https://github.com/flashbots/suave-geth/blob/b328d64689930a40eae0a6e834805f3feab6b58f/core/vm/contracts_suave.go#L237)
+- `dataId` (DataId): ID of the data record to retrieve
+- `key` (string): Key slot of the data to retrieve
 
-Address: `0x0000000000000000000000000000000042030000`
+Outputs:
+
+- `value` (bytes): Value of the data
+
+### `confidentialStore`
+
+Address: `0x0000000000000000000000000000000042020000`
 
-Initializes data records within the ConfidentialStore. `AllowedPeekers` specifies which addresses can "get" data. `AllowedStores` specifies which addresses can "set" data. Prior to storing data, all bids should undergo initialization via this precompile.
+Stores data in the confidential store. Requires the caller to be part of the `AllowedPeekers` for the associated data record.
 
 ```solidity
-function newDataRecord(uint64 decryptionCondition, address[] memory allowedPeekers, address[] memory allowedStores, string memory dataType)
+function confidentialStore(DataId dataId, string memory key, bytes memory value) internal view returns ()
 ```
 
-### `FetchDataRecords`
+Inputs:
 
-[🔗 Implementation](https://github.com/flashbots/suave-geth/blob/b328d64689930a40eae0a6e834805f3feab6b58f/core/vm/contracts_suave.go#L291)
+- `dataId` (DataId): ID of the data record to store
+- `key` (string): Key slot of the data to store
+- `value` (bytes): Value of the data to store
 
-Address: `0x0000000000000000000000000000000042030001`
+### `contextGet`
+
+Address: `0x0000000000000000000000000000000053300003`
 
-Retrieves all data records correlating with a specified decryption condition.
+Retrieves a value from the context
 
 ```solidity
-function fetchDataRecords(uint64 cond, string memory namespace) internal view returns (DataRecord[] memory)
+function contextGet(string memory key) internal view returns (bytes memory)
 ```
 
-### `EthCall`
+Inputs:
 
-[🔗 Implementation](https://github.com/flashbots/suave-geth/blob/b328d64689930a40eae0a6e834805f3feab6b58f/core/vm/contracts_suave_eth.go#L193)
+- `key` (string): Key of the value to retrieve
 
-Address: `0x0000000000000000000000000000000042100003`
+Outputs:
 
-Uses the `eth_call` JSON RPC method to let you simulate a function call and return the response.
+- `value` (bytes): Value of the key
+
+### `doHTTPRequest`
+
+Address: `0x0000000000000000000000000000000043200002`
+
+Performs an HTTP request and returns the response. `request` is the request to perform.
 
 ```solidity
- function ethcall(address contractAddr, bytes memory input1) internal view returns (bytes memory)
+function doHTTPRequest(HttpRequest memory request) internal view returns (bytes memory)
 ```
 
-### `SimulateBundle`
+Inputs:
 
-[🔗 Implementation](https://github.com/flashbots/suave-geth/blob/b328d64689930a40eae0a6e834805f3feab6b58f/core/vm/contracts_suave_eth.go#L115)
+- `request` (HttpRequest): Request to perform
```

Comment:
could probably give more info here - this whole section is repetitive without being informative